lock/src/seqlock.rs

```diff
@@ -62,6 +62,24 @@ impl RawSeqLock {
 
         seq == self.seq.load(Ordering::Relaxed)
     }
+
+    /// # Safety
+    ///
+    /// `seq` must be even.
+    pub unsafe fn upgrade(&self, seq: usize) -> Result<usize, ()> {
```

Comment:
I feel `Result<(), ()>` is sufficient, because the return value should coincide with the input `seq`.